server/app/com/xsn/explorer/services/MerchantnodeService.scala

```diff
@@ -0,0 +1,74 @@
+package com.xsn.explorer.services
+
+import com.alexitc.playsonify.core.FutureOr.Implicits.{FutureOps, OrOps}
+import com.alexitc.playsonify.core.{FutureApplicationResult, FuturePaginatedResult}
+import com.alexitc.playsonify.models.ordering.{FieldOrdering, OrderingCondition, OrderingQuery}
+import com.alexitc.playsonify.models.pagination.{Count, PaginatedQuery, PaginatedResult}
+import com.alexitc.playsonify.validators.PaginatedQueryValidator
+import com.xsn.explorer.errors.IPAddressFormatError
+import com.xsn.explorer.models.fields.MerchantnodeField
+import com.xsn.explorer.models.rpc.Merchantnode
+import com.xsn.explorer.models.values.IPAddress
+import com.xsn.explorer.parsers.MerchantnodeOrderingParser
+import javax.inject.Inject
+import org.scalactic.{Bad, Good}
+
+import scala.concurrent.ExecutionContext
+
+class MerchantnodeService @Inject()(
+    queryValidator: PaginatedQueryValidator,
+    merchantnodeOrderingParser: MerchantnodeOrderingParser,
+    xsnService: XSNService
+)(implicit ec: ExecutionContext) {
+  
+  def getMerchantnodes(
+      paginatedQuery: PaginatedQuery,
+      orderingQuery: OrderingQuery
+  ): FuturePaginatedResult[Merchantnode] = {
+    val result = for {
+      validatedQuery <- queryValidator.validate(paginatedQuery, 2000).toFutureOr
+      ordering <- merchantnodeOrderingParser.from(orderingQuery).toFutureOr
+      merchantnodes <- xsnService.getMerchantnodes().toFutureOr
```

Comment:
You need to do something quite similar to #185 to keep the merchant nodes in-memory.